pkg/promtail/server/server.go

```diff
@@ -156,9 +163,13 @@ func (s *Server) targets(rw http.ResponseWriter, _ *http.Request) {
 // ready serves the ready endpoint
 func (s *Server) ready(rw http.ResponseWriter, _ *http.Request) {
 	if s.tms.Ready() {
-		rw.WriteHeader(http.StatusNoContent)
+		rw.WriteHeader(http.StatusOK)
+		_, err := rw.Write(readinessProbeSuccess)
+		if err != nil {
+			level.Error(logutil.Logger).Log("msg", "error writing success message", "error", err)
+		}
 	} else {
-		rw.WriteHeader(http.StatusInternalServerError)
+		http.Error(rw, readinessProbeFailure, http.StatusInternalServerError)
```

Comment:
Ditto above - reduce nesting please.    See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 for discussion.